mock/mock.go

```diff
@@ -0,0 +1,16 @@
+package mock
+
+import "github.com/stretchr/testify/mock"
+
+type Client struct {
+	mock.Mock
+}
+
+func NewClient() *Client {
```

Comment:
The "normal" one returns an interface. The actual `client` type is hidden in the package (hence lowercase) - the idea being that only the interface is exposed in the public API - while giving a default implementation for consumers of the library.